src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -299,7 +300,13 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
             };
 
             if ($checkAttributes) {
-                foreach ($parameter->getAttributes(Autowire::class, \ReflectionAttribute::IS_INSTANCEOF) as $attribute) {
+                $attributes = array_merge($parameter->getAttributes(Autowire::class, \ReflectionAttribute::IS_INSTANCEOF), $parameter->getAttributes(Lazy::class, \ReflectionAttribute::IS_INSTANCEOF));
+
+                if (1 < \count($attributes)) {
```

Comment:
```suggestion
                if (!$attributes) {
```

Reply:
Can't, we don't want to check if it's empty, we want to check we have max 1 attribute, `Lazy` or `Autowire` but not both.